lib/core-integration/src/Test/Integration/Scenario/CLI/Shelley/Wallets.hs

```diff
@@ -456,18 +457,22 @@ spec = describe "SHELLEY_CLI_WALLETS" $ do
             else
                 err `shouldContain` errMsg400WalletIdEncoding
 
-    it "WALLETS_LIST_01 - Can list wallets" $ \ctx -> do
-        let name = "Wallet to be listed"
-        w1 <- emptyWalletWith' ctx (name, "secure-passphrase", 21)
-        _ <- emptyWallet' ctx
+    it "WALLETS_LIST_01 - Can list wallets" $ \ctx -> runResourceT $ do
+        let prefix = "WALLETS_LIST_01_can"
+        let name1 = prefix <> "#1"
+        let name2 = prefix <> "#2"
+        w1 <- emptyWalletWith' ctx (name1, "secure-passphrase", 21)
+        _ <- emptyWalletWith' ctx (name2, "secure-passphrase", 21)
         (Exit c, Stdout out, Stderr err) <- listWalletsViaCLI @t ctx
         c `shouldBe` ExitSuccess
         err `shouldBe` cmdOk
-        j <- expectValidJSON (Proxy @[ApiWallet]) out
+        let walFromThisTest w =
+                prefix `T.isPrefixOf` getWalletName (getApiT (view #name w))
+        j <- filter walFromThisTest <$> expectValidJSON (Proxy @[ApiWallet]) out
```

Comment:
Same check as before, but we name the wallets in this test with a unique prefix, such that we can filter out the rest.